### PR TITLE
[SDF-23] Transform class

### PR DESCRIPTION
**Pull request by kubazuch**
This should be merged **after** #7

### CODE REVIEW

File: libresin/libresin/core/transform.cpp [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+  if (!dirty_ || !inv_dirty_) {
+    for (const auto child : children_) {
+      child.get().mark_dirty();
+    }
+  }
+
+  dirty_ = inv_dirty_ = true;
+}
+
+void transform::clear_parent() {
```
Comment (migoox) [inline libresin/libresin/core/transform.cpp:149]: I would rename it to `remove_from_parent`

**Review by migoox — COMMENTED** [review]
I will read this PR again after getting some sleep:3

File: libresin/tests/core/transform_test.cpp [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+
+TEST_F(TransformTest, OrientationIsCalculatedProperly) {
+  // given
+  transform_.set_local_rot(glm::vec3(kPi / 2, 0, 0));
+  glm::vec3 front = transform_.front();
+  glm::vec3 right = transform_.right();
+  glm::vec3 up = transform_.up();
+
+  // when
+  std::println("{}", glm::to_string(transform_.local_to_world_matrix()));
```
Comment (migoox) [inline libresin/tests/core/transform_test.cpp:122]: I haven't noticed any prints in other tests so just asking if this print is accidental.

**Review by migoox — APPROVED** [review]
I slept quite a while 😴, great job!

File: libresin/tests/core/transform_test.cpp (hunk 2 above)
Comment (kubazuch) [reply libresin/tests/core/transform_test.cpp:122]: yessss 🙂‍↕️🙂‍↕️

Comment (kubazuch) [reply libresin/tests/core/transform_test.cpp:122]:
```suggestion
```

**Pull request merged**